client/signup/steps/woocommerce-install/step-store-address/index.tsx

```diff
@@ -75,6 +76,19 @@ export default function StepStoreAddress( props: WooCommerceInstallProps ): Reac
 		return onboardingProfile[ 'store_email' ] || '';
 	}
 
+	// Form validation.
+	const address1Error = getError( WOOCOMMERCE_STORE_ADDRESS_1 );
+	const address2Error = getError( WOOCOMMERCE_STORE_ADDRESS_2 );
+	const countryError = getError( WOOCOMMERCE_DEFAULT_COUNTRY );
+	const cityError = getError( WOOCOMMERCE_STORE_CITY );
+	const postcodeError = getError( WOOCOMMERCE_STORE_POSTCODE );
+	const emailError = getError( WOOCOMMERCE_ONBOARDING_PROFILE );
+
+	const [ submitted, setSubmitted ] = useState( 0 );
+	useEffect( () => {
+		return;
+	}, [ submitted ] );
```

Comment:
@allilevine this is a bit of a hack but fixes the re-render problem when validation runs. This should really be cleaned up next sprint but I think it will have to do for now.

Reply:
What's the re-render problem? It looks like the hook will run every time the `submitted` dependency is updated and won't do anything.

It might be worth looking into wrapping validation functions in a [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback)... maybe?

Reply:
The form wasn't re-rendering on submit, so the errors sometimes weren't displaying. I think by running the hook when `submittted` is updated, we're re-rendering on submit. But I would think that updating state (`submitted`) would be enough to re-render the component. 🤔